cli/src/generate/templates/.editorconfig

```diff
@@ -37,3 +34,6 @@ indent_size = 8
 [Makefile]
 indent_style = tab
 indent_size = 8
+
+[parser.c]
+indent_size = 2
```

Comment:
I think we need to add all of the generated headers too (`tree_sitter/{alloc,array,parser}.h`). Since almost all of the generated C code and most of the external scanners I have seen in the wild use 2-space indent, maybe it would make sense to just use `indent_size = 2` for all C/C++ here?